controller/api/views.py

```diff
@@ -210,6 +210,17 @@ def run(self, request, **kwargs):
         return Response(output_and_rc, status=status.HTTP_200_OK,
                         content_type='text/plain')
 
+    def update(self, request, **kwargs):
+        app = self.get_object()
+
+        if request.data.get('owner'):
+            if self.request.user != app.owner and not self.request.user.is_superuser:
+                raise PermissionDenied()
+            new_owner = get_object_or_404(User, username=request.data['owner'])
+            app.owner = new_owner
+        app.save()
+        return Response(status=status.HTTP_204_NO_CONTENT)
```

Comment:
A 204 means that we should never return a response body for the issuing request. A 200 could open this up by allowing us to send back more information about the request in a future enhancement to this endpoint (for example, if we were to return information about the new owner, the time it took to process the request etc.). For the sake of forwards compatibility, I think we should make this a 200 :)